realm/realm-library/src/main/java/io/realm/internal/TableView.java

```diff
@@ -510,8 +510,9 @@ public long upperBoundLong(long columnIndex, long value) {
     public TableView findAllLong(long columnIndex, long value){
         // Execute the disposal of abandoned realm objects each time a new realm object is created
         context.executeDelayedDisposal();
-        long nativeViewPtr = nativeFindAllInt(nativePtr, columnIndex, value);
-        try { 
+        long nativeViewPtr = 0;
+        try {
+            nativeViewPtr = nativeFindAllInt(nativePtr, columnIndex, value);
```

Comment:
```
JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeFindAllInt(
    JNIEnv* env, jobject, jlong nativeViewPtr, jlong columnIndex, jlong value)
{
    try {
        if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr) ||
            !COL_INDEX_AND_TYPE_VALID(env, TV(nativeViewPtr), columnIndex, type_Int))
            return 0;
        TableView* pResultView = new TableView( TV(nativeViewPtr)->find_all_int( S(columnIndex), value) );
        return reinterpret_cast<jlong>(pResultView);
    } CATCH_STD()
    return 0;
}
```
For the above code, only `new TableView` will throw. and when that throws, `nativeClose` doesn't have to be called and the `RealmException` will be thrown anyway.

Or do I misunderstand the purpose of this change?